src/map.rs

```diff
@@ -2421,10 +2425,7 @@ where
     where
         M: MapAccess<'de>,
     {
-        let map = match access.size_hint() {
-            Some(n) => HashMap::with_capacity(n),
-            None => HashMap::new(),
-        };
+        let map = S::default();
```

Comment:
You still probably want to use `HashMap::with_capacity_and_hasher` or `HashMap::with_hasher` here.

Reply:
Indeed! Just fixed this in 9822de1.